src/overlays/actors/ovl_Obj_Lift/z_obj_lift.c

```diff
@@ -30,29 +32,182 @@ const ActorInit Obj_Lift_InitVars = {
+    }
+}
+
+void ObjLift_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjLift* this = THIS;
+
+    func_80B960D8(this, globalCtx, &D_060003D0, 1);
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params >> 2) & 0x3F) != 0) {
+        Actor_Kill(&this->dyna.actor);
+        return;
+    }
+    Actor_SetScale(&this->dyna.actor, D_80B96A08[(this->dyna.actor.params >> 1) & 1]);
+    Actor_ProcessInitChain(&this->dyna.actor, D_80B969F4);
+    this->unk168.x = Rand_ZeroOne() * 65535.5f;
+    this->unk168.y = Rand_ZeroOne() * 65535.5f;
+    this->unk168.z = Rand_ZeroOne() * 65535.5f;
+    func_80B9651C(this);
+    osSyncPrintf("(Dungeon Lift)(arg_data 0x%04x)\n", this->dyna.actor.params);
+}
+
+void ObjLift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjLift* this = THIS;
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80B9651C(ObjLift* this) {
+    this->unk_16E = D_80B969C0[(this->dyna.actor.params >> 8) & 7];
+    func_80B960D0(this, &func_80B96560);
+}
+
+void func_80B96560(ObjLift* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 sp32;
+
+    if (func_8004356C(&this->dyna) != 0) {
+        if (this->unk_16E <= 0) {
+            if (((this->dyna.actor.params >> 8) & 7) == 7) {
+                func_80B967C0(this);
+            } else {
+                sp32 = Quake_Add(ACTIVE_CAM, 1);
+                Quake_SetSpeed(sp32, 10000);
+                Quake_SetQuakeValues(sp32, 2, 0, 0, 0);
+                Quake_SetCountdown(sp32, 20);
+                func_80B9664C(this);
+            }
+            return;
+        }
+    } else {
+        this->unk_16E = D_80B969C0[((this->dyna.actor.params >> 8) & 7)];
+    }
+}
+
+void func_80B9664C(ObjLift* this) {
+    this->unk_16E = 20;
+    func_80B960D0(this, func_80B96678);
+}
+
+void func_80B96678(ObjLift* this, GlobalContext* globalCtx) {
+    if ((s32)this->unk_16E <= 0) {
+        func_80B967C0(this);
+    } else {
+        this->unk168.x = this->unk168.x + 10000;
+        this->dyna.actor.world.rot.x = (s16)(Math_SinS(this->unk168.x) * 300.0f) + this->dyna.actor.home.rot.x;
+        this->dyna.actor.world.rot.z = (s16)(Math_CosS(this->unk168.x) * 300.0f) + this->dyna.actor.home.rot.z;
+        this->dyna.actor.shape.rot.x = this->dyna.actor.world.rot.x;
+        this->dyna.actor.shape.rot.z = this->dyna.actor.world.rot.z;
+        this->unk168.y = this->unk168.y + 18000;
+        this->dyna.actor.world.pos.y = Math_SinS(this->unk168.y) + this->dyna.actor.home.pos.y;
+        this->unk168.z = this->unk168.z + 18000;
+        this->dyna.actor.world.pos.x = (Math_SinS(this->unk168.z) * 3.0f) + this->dyna.actor.home.pos.x;
+        this->dyna.actor.world.pos.z = (Math_CosS(this->unk168.z) * 3.0f) + this->dyna.actor.home.pos.z;
+    }
+    if ((this->unk_16E & 3) == 3) {
+        Audio_PlaySoundAtPosition(globalCtx, (Vec3f*)&this->dyna.actor.world, 0x10, (u16)0x2838U);
```

Comment:
```suggestion
        Audio_PlaySoundAtPosition(globalCtx, &this->dyna.actor.world.pos, 0x10, NA_SE_EV_BLOCK_SHAKE);
```
Want `world.pos` rather than casting, we have defines for sfx ids.